psi4/src/psi4/libpsipe/psipe.cc

```diff
@@ -0,0 +1,259 @@
+/*
+ * @BEGIN LICENSE
+ *
+ * Psi4: an open-source quantum chemistry software package
+ *
+ * Copyright (c) 2007-2018 The Psi4 Developers.
+ *
+ * The copyrights for code used from other parties are included in
+ * the corresponding files.
+ *
+ * This file is part of Psi4.
+ *
+ * Psi4 is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * Psi4 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with Psi4; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * @END LICENSE
+ */
+#ifdef USING_cppe
+
+#include <iomanip>
+
+#include <armadillo>
```

Comment:
Where is armadillo coming from in the build process? How onerous would it be to switch out for LAPACK calls or `psi4.core.Matrix`?

Reply:
When the CPPE library is built, cmake searches for armadillo on the system, and builds it from source if it's not found ([here is the code in my repo](https://github.com/maxscheurer/cppe/blob/master/external/upstream/armadillo/CMakeLists.txt)). The stand-alone CPPE build works fine on Travis CI at least.

I think at the moment it is very hard to replace armadillo since this would require re-writing a major part of the library, but I understand that having yet another linear algebra library
is annoying...

Just some additional information: I originally developed the CPPE library for PE in Q-Chem where armadillo commonly used, so it was kind of convenient to use it for CPPE, too.